### PR TITLE
Deal with a cancellation exception being t hrown through a sync-path in NavigateTo

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/Diagnostics/SpellCheck/SpellCheckTests.cs [hunk 1]
```diff
@@ -7,6 +7,8 @@
 using Microsoft.CodeAnalysis.CSharp.CodeFixes.Spellcheck;
 using Xunit;
 using Roslyn.Test.Utilities;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Collections.Generic;
```
Comment (DustinCampbell) [inline src/EditorFeatures/CSharpTest/Diagnostics/SpellCheck/SpellCheckTests.cs]: sort usings

File: src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.Searcher.cs [hunk 2]
```diff
@@ -68,6 +68,9 @@ private class Searcher
                         await Task.WhenAll(searchTasks).ConfigureAwait(false);
                     }
                 }
+                catch (OperationCanceledException)
```
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.Searcher.cs]: Why is `Search()` an async void method?

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.Searcher.cs]: It's basically an event handler.  The API between us and VS is that they simply call 'search', and that search is supposed to kick off work and then callback when it's done.  Hence 'void'.  however, we want to be able to use await's in our code so we don't have to kick off work and then do annoying .ContinueWiths.  Hence 'async'.  so 'async void' it is :-/

**Pull request merged**